src/integrationTest/sam.test.ts

```diff
@@ -0,0 +1,192 @@
+}
+
+// Iterate through and test all runtimes
+for (const runtime of runtimes) {
+    describe(`SAM Integration tests ${runtime.name}`, async () => {
+        before(async function() {
+            // tslint:disable-next-line: no-invalid-this
+            this.timeout(TIMEOUT)
+            projectSDK = runtime.name
+            projectPath = runtime.path
+            debuggerType = runtime.debuggerType
+            // set up debug config
+            vscode.debug.onDidChangeActiveDebugSession(onDebugChanged)
+            await activateExtension('amazonwebservices.aws-toolkit-vscode')
+            console.log(`Using SDK ${projectSDK} with project in path ${projectPath}`)
+            tryRemoveProjectFolder()
+            mkdirpSync(projectFolder)
+            // this is really test 1, but since it has to run before everything it's in the before section
+            const initArguments: SamCliInitArgs = {
+                name: 'testProject',
+                location: projectFolder,
+                runtime: projectSDK as SamLambdaRuntime
+            }
+            const samCliContext = getSamCliContext()
+            await runSamCliInit(initArguments, samCliContext.invoker)
+            // Activate the relevent extensions if needed
+            if (projectSDK.includes('dotnet')) {
+                await activateExtension('ms-vscode.csharp')
+            }
+            if (projectSDK.includes('python')) {
+                await activateExtension('ms-python.python')
+            }
+            documentUri = await openSamProject()
+        })
+
+        after(async () => {
+            tryRemoveProjectFolder()
+        })
+
+        it('Generates a template with a proper runtime', async () => {
+            const fileContents = readFileSync(`${projectFolder}/testProject/template.yaml`).toString()
+            assert.ok(fileContents.includes(`Runtime: ${projectSDK}`))
+        })
+
+        it('Fails to create template when it already exists', async () => {
+            const initArguments: SamCliInitArgs = {
+                name: 'testProject',
+                location: projectFolder,
+                runtime: projectSDK as SamLambdaRuntime
+            }
+            console.log(initArguments.location)
+            const samCliContext = getSamCliContext()
+            await runSamCliInit(initArguments, samCliContext.invoker).catch((e: Error) => {
+                assert(e.message.includes('directory already exists'))
+            })
+        }).timeout(TIMEOUT)
+
+        it('Invokes the run codelens', async () => {
+            const [runCodeLens] = await getCodeLensesOrTimeout()
+            assert.ok(runCodeLens.command)
+            const command = runCodeLens.command!
+            assert.ok(command.arguments)
+            const runResult: any | undefined = await vscode.commands.executeCommand(
+                command.command,
+                ...command.arguments!
+            )
+            assert.ok(runResult)
```

Comment:
Your validations against runResult are identical in both tests (the difference being the expected metadata-debug value). Consider creating an `assertLocalInvokeCommandResults` method.

Reply:
sure made a fucntion